lib/cpp/test/TMemoryBufferTest.cpp

```diff
@@ -117,4 +117,17 @@ BOOST_AUTO_TEST_CASE(test_exceptions) {
   BOOST_CHECK_NO_THROW(buf2.write((const uint8_t*)"bar", 3));
 }
 
+#ifndef _WIN32
+// We can't allocate 1 GB of memory in 32-bit environments.
```

Comment:
We'd have to write in a loop to trigger buffer resizing if I understand correctly. It'd save the vector allocation in the test, but `TMemoryBuffer` would still grow. I agree a 50% reduction would be better than nothing, for sure.